testpar/t_subfiling_vfd.c

```diff
@@ -275,6 +298,599 @@ test_create_and_close(void)
+                        }
+                        free(buf);
+                    }
+
+                    if (fspace_id >= 0)
+                        H5Sclose(fspace_id);
+                }
+                H5Dclose(dset_id);
+            }
+
+            H5Fclose(sec2_file_id);
+        }
+        else {
+            printf("ERROR: Could not re-open fused file with sec2 driver for data validation\n");
+            ret_value = FAIL;
+            goto done;
+        }
+    }
+
+done:
+    if (h5fuse_cmd)
+        free(h5fuse_cmd);
+
+    H5E_BEGIN_TRY
+    {
+        if (fused_file_id >= 0)
+            H5Fclose(fused_file_id);
+    }
+    H5E_END_TRY;
+
+    return ret_value;
+}
+
+/* Helper function to find configuration file */
+static char *
+find_config_file(const char *main_filename)
+{
+    h5_stat_t file_info;
+    char     *config_filename = NULL;
+
+    /* Get file inode for config filename construction */
+    if (HDstat(main_filename, &file_info) < 0) {
+        return NULL;
+    }
+
+    config_filename = malloc(PATH_MAX);
+    if (!config_filename) {
+        return NULL;
+    }
+
+    /* Construct expected config filename using the same pattern as the main code */
+    snprintf(config_filename, PATH_MAX, "%s/" H5FD_SUBFILING_CONFIG_FILENAME_TEMPLATE,
+             config_dir ? config_dir : ".", main_filename, (uint64_t)file_info.st_ino);
+
+    /* Check if config file exists */
+    if (access(config_filename, F_OK) == 0) {
+        return config_filename;
+    }
+
+    /* Try alternative locations */
+    snprintf(config_filename, PATH_MAX, "./%s.config", main_filename);
+    if (access(config_filename, F_OK) == 0) {
+        return config_filename;
+    }
+
+    /* Search for any .config file in current directory */
+    snprintf(config_filename, PATH_MAX, "./subfiling.config");
+    if (access(config_filename, F_OK) == 0) {
+        return config_filename;
+    }
+
+    free(config_filename);
+    return NULL;
+}
+
+/* Helper function to create FAPL with specific IOC selection and count */
+static hid_t
+create_subfiling_fapl_with_ioc_selection_and_count(H5FD_subfiling_ioc_select_t selection_type, int ioc_count)
+{
+    hid_t                   fapl_id = H5I_INVALID_HID;
+    H5FD_subfiling_config_t subf_config;
+    herr_t                  ret;
+
+    /* Create base FAPL with MPI settings */
+    fapl_id = create_subfiling_ioc_fapl(comm_g, info_g, false, NULL, 0);
+    VRFY((fapl_id >= 0), "Base FAPL creation succeeded");
+
+    /* Get current subfiling configuration */
+    ret = H5Pget_fapl_subfiling(fapl_id, &subf_config);
+    VRFY((ret >= 0), "H5Pget_fapl_subfiling succeeded");
+
+    /* Modify IOC selection method and stripe count */
+    subf_config.shared_cfg.ioc_selection = selection_type;
+    subf_config.shared_cfg.stripe_count  = ioc_count;
+
+    /* Apply modified configuration */
+    ret = H5Pset_fapl_subfiling(fapl_id, &subf_config);
+    VRFY((ret >= 0), "H5Pset_fapl_subfiling succeeded");
```

Comment:
`subf_config.ioc_fapl_id` probably needs closed here, otherwise the ID will leak and it could result in usage of MPI routines after MPI has been finalized.

Reply:
fixed